### PR TITLE
Add CODEOWNERS and restricted access datasets documentation

### PR DESCRIPTION
**Pull request by whd**
I'm open to any feedback on better or additional places to put this info as well as content.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,3 +4,6 @@ Checklist for reviewer:
 - [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
 in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
 - [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated
+
+For modifications to schemas in restricted namespaces (see `CODEOWNERS`:
```
Comment (jklukas) [inline .github/pull_request_template.md:8]:
```suggestion
For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -4,3 +4,6 @@ Checklist for reviewer:
...
+- [ ] Follow the change control procedure in https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck
```
Comment (jklukas) [inline .github/pull_request_template.md:9]:
```suggestion
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)
```

File: README.md [hunk 3]
```diff
@@ -420,6 +420,12 @@ Contributing
 
 When adding or modifying a query in this repository, make your changes in the `sql/` directory.
 
+If updating schemas associated with certain restricted-access datasets (specified in
+`CODEOWNERS`), a CODEOWNER (usually SRE) will automatically be assigned to review the PR. Please
+follow [these additional instructions](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)
```
Comment (jklukas) [inline README.md:425]:
```suggestion
follow [change control procedures](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)
```

**Review by jklukas — COMMENTED** [review]
These seem like reasonable spots for the documentation. I will add an item to discuss these changes in data-platform-infra-wg next week.

**Comment by wlach** [comment]
> These seem like reasonable spots for the documentation. I will add an item to discuss these changes in data-platform-infra-wg next week.

When #2527 lands, the right place for the docs is IMO `CONTRIBUTING.md`.

**Review by jklukas — APPROVED** [review]

**Pull request merged**